### DISCUSSION TITLE
Managed BA ExecuteFilesInUseEventArgs::Files varying formats?

### DISCUSSION THREAD

**Discussion by WillR27** [wixtoolset/issues#6957] [Questions]
This is using WiX 3.11.2.

I am trying to access the processes that are locking the files when executing on an MSI package. Using the example snippet below I have managed to get 3 differently formatted outputs from the same MSI/BA.

```csharp
private void OnExecuteFilesInUse(object sender, ExecuteFilesInUseEventArgs e)
{
    Engine.Log(LogLevel.Standard, $"Files in use for {e.PackageId}: {string.Join(", ", e.Files)}");
}
```

90% of the time I get the following where it lists the window names(?), but also a lot of empty strings (empty window names(?)):

_[0F70:2DBC][2022-10-12T09:41:01]i000: Files in use for Package1.msi: , WindowName1, WindowName2, , , , , , , , , , , , , , , , , , , , , , , , , , , ,_

The rest of the time it is one of the following:

_[0E0C:29BC][2022-10-07T08:40:33]i000: Files in use for Package1.msi: 7188, Process1, 5564, AnotherName, 11644, SomeOtherProcess, 6092, SomeOtherProcess, 11924, SomeOtherProcess, 8044, SomeOtherProcess, 11060, SomeOtherProcess, ,

[0E0C:29BC][2022-10-07T08:40:33]i000: Files in use for Package1.msi: Process1, Process1 (Process Id: 7188), AnotherName, AnotherName (Process Id: 5564), SomeOtherProcess, SomeOtherProcess (Process Id: 11644), SomeOtherProcess, SomeOtherProcess (Process Id: 6092), SomeOtherProcess, SomeOtherProcess (Process Id: 11924), SomeOtherProcess, SomeOtherProcess (Process Id: 8044), SomeOtherProcess, SomeOtherProcess (Process Id: 11060)_

I'm just wondering what causes the different results for `e.Files` and whether there is any documentation I have missed? I can't find any other mention of this. Thanks.

**Comment by robmen** [comment]
I _think_ this is complexity the Windows Installer exposes due to FilesInUse and MsiRmFilesInUse. Burn didn't abstract this away in v3 much. I don't remember if we made any progress in v4 (though I do think we discussed it a bit).
